### DISCUSSION TITLE
Synchronizing seperate Spring animations

### DISCUSSION THREAD

**Discussion by Eunanibus** [pmndrs/react-spring#1594] [Support]
Hey folks,

I've been at this for about 48 hours now and I've still not found a solution, so posting here is my last attempt before I move on to another way to solve this.

I'm creating a login form that animates. The form works in a way similar to this:

<img src='https://media.giphy.com/media/1rK9pvvWZUBReNAt56/giphy.gif' />

As a result, I have two entirely separate divs that need to be animated with entirely different values.

So my attempt to do this was with two individual `useSpring()` instantiations. My assumption was that without `useChain`, these animations would execute together. What I found was that it was sequential instead. These are all dependent on a single boolean state value, but they still execute in sequence (from parent to child).

Provided I have several divs, all that need to be animated slightly differently; How do I go about synchronizing them? The examples in the documentation only provide references to lists which isn't exactly what I'm trying to achieve here.

Any advice?

Thanks

**Comment by slapxxi** [comment]
It's kind of rough, but it's a possible solution I guess: [Demo](https://codesandbox.io/s/distracted-darkness-l4t2d?file=/src/index.tsx)

**Comment by GuiShenHe** [comment]
@Eunanibus Did you ever figure out how to go about this? Previous versions of popmotion offered this but has long since abandoned this.

**Reply by Eunanibus** [reply]
Sorry @GuiShenHe  - Yes I did manage it, I can upload an example to a repo for you if you'd like?

**Reply by GuiShenHe** [reply]
That'd be great. Thanks!

**Comment by puopg** [comment]
For anyone coming here who is curious how to synchronize multiple springs, this is what the `useSprings` hook is designed for. THe API I must say is kinda weird, but essentially you give it a number of animations you want, and then a call back to return the "styles" for each animation index. 

This gives you back an array of `springStyles` that you can index into for the animation you want for some element, and then you can apply that individual style to the element. 

When you want to perform multiple animations, you need to give another callback that dictates the desired styles for each animation that you want to perform.
